src/main/java/org/zalando/fahrschein/NakadiReader.java

```diff
@@ -240,22 +249,30 @@ void runInternal(long timeout, TimeUnit timeoutUnit) throws IOException, Backoff
 
                 LOG.debug("Cursor for [{}] partition [{}] at offset [{}]", eventName, cursor.getPartition(), cursor.getOffset());
 
+                metricsCollector.ifPresent(mc -> mc.markBatchesReceived());
+
                 final JsonToken token = jsonParser.nextToken();
                 if (token != JsonToken.END_OBJECT) {
                     expectField(jsonParser, "events");
 
                     final List<T> events = readEvents(jsonParser);
+                    metricsCollector.ifPresent(mc -> mc.markEventsReceived(events.size()));
 
                     expectToken(jsonParser, JsonToken.END_OBJECT);
 
                     final Batch<T> batch = new Batch<>(cursor, Collections.unmodifiableList(events));
 
                     processBatch(batch);
+                } else {
+                    // it's a keep alive batch
+                    metricsCollector.ifPresent(mc -> mc.markEventsReceived(0));
```

Comment:
Should the keep alive message count as "events received"? Also, "events received" does not tell us that those events were processed successfully. I would propose the following meters instead:

 - messages received (instead of batches received), this includes keep-alive messages
 - events processed successfully
 - messages failed
 - maybe messages processed (messages received should then equal messages failed + messages processed)

What do you think?